[qt5-base:x64-linux] build failure: [libpq] not found with dynamic linkage

**Host Environment**
- OS: [Linux - Mint 20 - Ubuntu 20.04]
- Compiler: gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

**To Reproduce**
Steps to reproduce the behavior:
`git clean -dfx`
`git reset --hard origin/master`
`./bootstrap-vcpkg.sh`
`./vcpkg install --overlay-triplets=/home/xxx/custom-triplets qt5\[essentials\]`

> "custom" tripplet `x64-linux.cmake`:
```cmake
set(VCPKG_TARGET_ARCHITECTURE x64)
set(VCPKG_CRT_LINKAGE dynamic)
set(VCPKG_LIBRARY_LINKAGE dynamic)

set(VCPKG_CMAKE_SYSTEM_NAME Linux)
```

**Failure logs** 
- console:
```
-- Figuring out qt target mkspec. Target arch x64
-- Target mkspec set to: linux-g++
-- Host mkspec not set. Qt's own buildsystem will try to figure out the host system
-- Using cached /home/dweckmann/Work/vcpkg/downloads/qtbase-everywhere-src-5.15.1.tar.xz
-- Cleaning sources at /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/src/5.15.1-63a9328040.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/dweckmann/Work/vcpkg/downloads/qtbase-everywhere-src-5.15.1.tar.xz
-- Applying patch patches/winmain_pro.patch
-- Applying patch patches/windows_prf.patch
-- Applying patch patches/qt_app.patch
-- Applying patch patches/gui_configure.patch
-- Applying patch patches/icu.patch
-- Applying patch patches/xlib.patch
-- Applying patch patches/egl.patch
-- Applying patch patches/zstdd.patch
-- Applying patch patches/Qt5BasicConfig.patch
-- Applying patch patches/Qt5PluginTarget.patch
-- Applying patch patches/create_cmake.patch
-- Applying patch patches/Qt5GuiConfigExtras.patch
-- Using source at /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/src/5.15.1-63a9328040.clean
-- Configuring x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:106 (message):
    Command failed: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/src/5.15.1-63a9328040.clean/configure -confirm-license -opensource -verbose -system-zlib -system-libjpeg -system-libpng -system-freetype -system-pcre -system-doubleconversion -system-sqlite -system-harfbuzz -sql-psql -icu -no-angle -fontconfig -xcb-xlib -xcb -linuxfb -nomake examples -nomake tests LIBJPEG_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libjpeg.so ZLIB_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libz.so "LIBPNG_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libz.so" PCRE2_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpcre2-16.so "FREETYPE_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libfreetyped.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbz2d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libz.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbrotlidec.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbrotlicommon.so" "ICU_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libicui18n.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libicutu.so  /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libicuuc.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libicuio.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libicudata.so" QMAKE_LIBS_PRIVATE+=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbz2d.so QMAKE_LIBS_PRIVATE+=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so "PSQL_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpq.so PSQL_PORT_DEBUG-NOTFOUND  PSQL_COMMON_DEBUG-NOTFOUND /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libssl.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libcrypto.so -ldl -lpthread" "SQLITE_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libsqlite3.so -ldl -lpthread" "HARFBUZZ_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libharfbuzz.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libfreetyped.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbz2d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libz.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbrotlidec.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbrotlicommon.so  -lpthread" "OPENSSL_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libssl.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libcrypto.so -ldl -lpthread" "FONTCONFIG_LIBS=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libfontconfig.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libfreetyped.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libexpat.so -luuid" -debug -prefix /home/dweckmann/Work/vcpkg/installed/x64-linux -hostprefix /home/dweckmann/Work/vcpkg/installed/x64-linux/tools/qt5/debug -hostlibdir /home/dweckmann/Work/vcpkg/installed/x64-linux/tools/qt5/debug/lib -hostbindir /home/dweckmann/Work/vcpkg/installed/x64-linux/tools/qt5/debug/bin -archdatadir /home/dweckmann/Work/vcpkg/installed/x64-linux/tools/qt5/debug -datadir /home/dweckmann/Work/vcpkg/installed/x64-linux/share/qt5/debug -plugindir /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/plugins -qmldir /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/qml -headerdir /home/dweckmann/Work/vcpkg/installed/x64-linux/include -libexecdir /home/dweckmann/Work/vcpkg/installed/x64-linux/tools/qt5/debug -bindir /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/bin -libdir /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib -I /home/dweckmann/Work/vcpkg/installed/x64-linux/include -L /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib -L /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/manual-link -platform linux-g++
    Working Directory: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg
    Error code: 3
    See logs for more information:
      /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/config-x64-linux-dbg-out.log
      /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/config-x64-linux-dbg-err.log

Call Stack (most recent call first):
  ports/qt5-base/cmake/configure_qt.cmake:126 (vcpkg_execute_required_process)
  ports/qt5-base/portfile.cmake:302 (configure_qt)
  scripts/ports.cmake:136 (include)


Error: Building package qt5-base:x64-linux failed with: BUILD_FAILED
```

- config-x64-linux-dbg-err.log:
```
...
ERROR: Feature 'sql-psql' was enabled, but the pre-condition 'libs.psql' failed.
...
```

- config.log:
```
looking for library psql
Trying source 0 (type pkgConfig) of library psql ...
+ /usr/bin/pkg-config --exists --silence-errors libpq
pkg-config did not find package.
  => source produced no result.
Trying source 1 (type psqlConfig) of library psql ...
pg_config not found.
  => source produced no result.
Trying source 2 (type psqlEnv) of library psql ...
  => source failed condition 'config.win32'.
Trying source 3 (type psqlEnv) of library psql ...
+ cd /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/config.tests/psql && /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/bin/qmake "CONFIG -= qt debug_and_release app_bundle lib_bundle" "CONFIG += shared warn_off console single_arch" 'QMAKE_LIBDIR += /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/manual-link' 'INCLUDEPATH += /home/dweckmann/Work/vcpkg/installed/x64-linux/include' QMAKE_LIBS_PRIVATE+=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbz2d.so QMAKE_LIBS_PRIVATE+=/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so 'QMAKE_USE += psql' 'QMAKE_LIBS_PSQL = /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpq.so PSQL_PORT_DEBUG-NOTFOUND PSQL_COMMON_DEBUG-NOTFOUND /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libssl.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libcrypto.so -ldl -lpthread' /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/config.tests/psql
+ cd /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/config.tests/psql && MAKEFLAGS= /usr/bin/make
> g++ -c -pipe -g -w -fPIC  -I. -I/home/dweckmann/Work/vcpkg/installed/x64-linux/include -I/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/src/5.15.1-63a9328040.clean/mkspecs/linux-g++ -o main.o main.cpp
> g++  -o psql main.o   -L/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib -L/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/manual-link /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpq.so PSQL_PORT_DEBUG-NOTFOUND PSQL_COMMON_DEBUG-NOTFOUND /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libssl.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libcrypto.so -ldl -lpthread   /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbz2d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so
> g++: error: PSQL_PORT_DEBUG-NOTFOUND: Aucun fichier ou dossier de ce type
> g++: error: PSQL_COMMON_DEBUG-NOTFOUND: Aucun fichier ou dossier de ce type
> make: *** [Makefile:68 : psql] Erreur 1
 => source failed verification.
test config.qtbase_sqldrivers.libraries.psql FAILED
```
[config-x64-linux-dbg-err.log](https://github.com/microsoft/vcpkg/files/5622139/config-x64-linux-dbg-err.log)
[config.log](https://github.com/microsoft/vcpkg/files/5622146/config.log)

Did I something wrong ?
Seems it found the release psql, but not the debug one.

Well, sorry, the configure step pass, but not the build itself. There are undefined symbols (zstd and icu?).

```
/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/src/5.15.1-63a9328040.clean/src/corelib/kernel/qobjectdefs.h:0: Note: No relevant classes found. No output generated.
/usr/bin/ld : avertissement : libicui18n.so.67, requis par /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so, non trouvé (essayez avec -rpath ou -rpath-link)
/usr/bin/ld : avertissement : libicuuc.so.67, requis par /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so, non trouvé (essayez avec -rpath ou -rpath-link)
/usr/bin/ld : avertissement : libzstdd.so.1, requis par /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so, non trouvé (essayez avec -rpath ou -rpath-link)
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_inDaylightTime_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_toUnicode_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_getDSTSavings_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_setMillis_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_getAvailableName_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_compareNames_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « u_strToLower_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « uenum_close_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_getStandardName_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_close_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_fromUnicode_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucol_strcoll_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_setSubstChars_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_getAlias_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_clone_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_open_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_getMaxCharSize_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ZSTD_isError »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ZSTD_getErrorName »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « uenum_next_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucol_getSortKey_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_get_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucol_open_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « u_strToUpper_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_getDefaultTimeZone_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_openTimeZoneIDEnumeration_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_countAvailable_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucol_close_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_open_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ZSTD_decompress »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « u_errorName_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_getTimeZoneDisplayName_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_openCountryTimeZones_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucol_setAttribute_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_openTimeZones_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ZSTD_getFrameContentSize »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_countAliases_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucal_close_67 »
/usr/bin/ld : /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so : référence indéfinie vers « ucnv_getDefaultName_67 »
collect2: error: ld returned 1 exit status

```

Well, in fact the log doesn't print the right error, when compiling by hand invoking make, the build fail on:

```bash
make[1]: Entering directory '/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/src/tools/qlalr'
g++ -Wl,--enable-new-dtags -Wl,-z,origin -Wl,-rpath,\$ORIGIN/../../../../debug/lib -o ../../../bin/qlalr .obj/compress.o .obj/cppgenerator.o .obj/dotgraph.o .obj/lalr.o .obj/main.o .obj/parsetable.o .obj/recognizer.o .obj/grammar.o   -L/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib -L/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/manual-link /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so -lpthread   /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libbz2d.so /home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib/libpng16d.so
/usr/bin/ld: warning: libicui18n.so.67, needed by /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libicuuc.so.67, needed by /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: warning: libzstdd.so.1, needed by /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so, not found (try using -rpath or -rpath-link)
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_inDaylightTime_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_toUnicode_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_getDSTSavings_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_setMillis_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_getAvailableName_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_compareNames_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `u_strToLower_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `uenum_close_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_getStandardName_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_close_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_fromUnicode_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucol_strcoll_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_setSubstChars_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_getAlias_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_clone_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_open_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_getMaxCharSize_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ZSTD_isError'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ZSTD_getErrorName'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `uenum_next_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucol_getSortKey_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_get_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucol_open_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `u_strToUpper_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_getDefaultTimeZone_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_openTimeZoneIDEnumeration_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_countAvailable_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucol_close_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_open_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ZSTD_decompress'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `u_errorName_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_getTimeZoneDisplayName_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_openCountryTimeZones_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucol_setAttribute_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_openTimeZones_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ZSTD_getFrameContentSize'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_countAliases_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucal_close_67'
/usr/bin/ld: /home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/lib/libQt5Core.so: undefined reference to `ucnv_getDefaultName_67'
collect2: error: ld returned 1 exit status
make[1]: *** [Makefile:138: ../../../bin/qlalr] Error 1
make[1]: Leaving directory '/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/src/tools/qlalr'
make: *** [Makefile:196: sub-qlalr-make_first] Error 2
```

Although the `-L/home/dweckmann/Work/vcpkg/installed/x64-linux/debug/lib' is really there, ld doesn't find this 3 libs. By adding the full path to them (and their dependencies..) it obviously works, but -L should be enough...

If someone have a clue...

Good idea, it get a little bit further (build is successful), unfortunately, the use of tool like `rcc` or `qvkgen` fails miserably:

```bash
/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/bin/rcc: error while loading shared libraries: libzstdd.so.1: cannot open shared object file: No such file or directory
/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/bin/rcc: error while loading shared libraries: libzstdd.so.1: cannot open shared object file: No such file or directory
/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/x64-linux-dbg/bin/qvkgen: error while loading shared libraries: libicui18n.so.67: cannot open shared object file: No such file or directory
make[2]: *** [Makefile:2188 : vulkan/qvulkanfunctions.h] Erreur 127
make[1]: *** [Makefile:450 : sub-gui-make_first] Erreur 2
make[1]: *** Attente des tâches non terminées....
/home/dweckmann/Work/vcpkg/buildtrees/qt5-base/src/5.15.1-63a9328040.clean/src/network/access/qnetworkaccessdebugpipebackend_p.h:0: Note: No relevant classes found. No output generated.
make: *** [Makefile:51 : sub-src-make_first] Erreur 2
```

Of course setting LD_LIBRARY_PATH will work, but I doubt it should be done that way.

Anyway, thanks for your help, I will dig up a bit later. I'm really surprised that I'm one of the few who build Qt with dynamic linking on Linux. Building Qt in static basically make your application GPL only and it takes ages to link it, not very pleasant experience when debuging.

--- a/ports/qt5-base/CONTROL
+++ b/ports/qt5-base/CONTROL
@@ -1,6 +1,6 @@
 Source: qt5-base
 Version: 5.15.1
-Port-Version: 2
+Port-Version: 3
 Homepage: https://www.qt.io/
 Description: Qt5 Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.
 Build-Depends: zlib, zstd, libjpeg-turbo, libpng, freetype, pcre2, harfbuzz, sqlite3, libpq, double-conversion, openssl, angle (!windows), egl-registry, icu (!uwp), fontconfig (!windows)

--- a/ports/qt5-base/cmake/install_qt.cmake
+++ b/ports/qt5-base/cmake/install_qt.cmake
@@ -60,7 +60,23 @@ function(install_qt)
     foreach(_buildname ${BUILDTYPES})
         set(_build_triplet ${TARGET_TRIPLET}-${_short_name_${_buildname}})
 
-        vcpkg_add_to_path(PREPEND "${CURRENT_INSTALLED_DIR}${_path_suffix_${_buildname}}/bin")
+        set(_installed_prefix_ "${CURRENT_INSTALLED_DIR}${_path_suffix_${_buildname}}")
+        set(_installed_libpath_ "${_installed_prefix_}/lib/${VCPKG_HOST_PATH_SEPARATOR}${_installed_prefix_}/lib/manual-link/")
+
+        vcpkg_add_to_path(PREPEND "${_installed_prefix_}/bin")
+        vcpkg_add_to_path(PREPEND "${_installed_prefix_}/lib")
+
+        # We set LD_LIBRARY_PATH ENV variable to allow executing Qt tools (rcc,...) even with dynamic linking
+        if(CMAKE_HOST_UNIX)
+            if(DEFINED ENV{LD_LIBRARY_PATH})
+                set(_ld_library_path_defined_ TRUE)
+                set(_ld_library_path_backup_ $ENV{LD_LIBRARY_PATH})
+                set(ENV{LD_LIBRARY_PATH} "${_installed_libpath_}${VCPKG_HOST_PATH_SEPARATOR}${_ld_library_path_backup_}")
+            else()
+                set(_ld_library_path_defined_ FALSE)
+                set(ENV{LD_LIBRARY_PATH} "${_installed_libpath_}")
+            endif()
+        endif()
 
         if(VCPKG_TARGET_IS_OSX)
            # For some reason there will be an error on MacOSX without this clean!
@@ -100,5 +116,14 @@ function(install_qt)
         )
         message(STATUS "Package ${_build_triplet} done")
         set(ENV{PATH} "${_path}")
+
+        # Restore backup
+        if(CMAKE_HOST_UNIX)
+            if(_ld_library_path_defined_)
+                set(ENV{LD_LIBRARY_PATH} "${_ld_library_path_backup_}")                
+            else()
+                unset(ENV{LD_LIBRARY_PATH})
+            endif()
+        endif()
     endforeach()
 endfunction()

--- a/ports/qt5-base/portfile.cmake
+++ b/ports/qt5-base/portfile.cmake
@@ -120,10 +120,14 @@ find_library(LIBPNG_RELEASE NAMES png16 libpng16 PATHS "${CURRENT_INSTALLED_DIR}
 find_library(LIBPNG_DEBUG NAMES png16 png16d libpng16 libpng16d PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
 find_library(PSQL_RELEASE NAMES pq libpq PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
 find_library(PSQL_DEBUG NAMES pq libpq pqd libpqd PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
-find_library(PSQL_COMMON_RELEASE NAMES pgcommon libpgcommon PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
-find_library(PSQL_COMMON_DEBUG NAMES pgcommon libpgcommon pgcommond libpgcommond PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
-find_library(PSQL_PORT_RELEASE NAMES pgport libpgport PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
-find_library(PSQL_PORT_DEBUG NAMES pgport libpgport pgportd libpgportd PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
+
+if(NOT (PSQL_RELEASE MATCHES ".*\.so") AND NOT (PSQL_DEBUG MATCHES ".*\.so"))
+    find_library(PSQL_COMMON_RELEASE NAMES pgcommon libpgcommon PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
+    find_library(PSQL_COMMON_DEBUG NAMES pgcommon libpgcommon pgcommond libpgcommond PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
+    find_library(PSQL_PORT_RELEASE NAMES pgport libpgport PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) # Depends on openssl and zlib(linux)
+    find_library(PSQL_PORT_DEBUG NAMES pgport libpgport pgportd libpgportd PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
+endif()
+
 find_library(PCRE2_RELEASE NAMES pcre2-16 PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
 find_library(PCRE2_DEBUG NAMES pcre2-16 pcre2-16d PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
 find_library(FREETYPE_RELEASE NAMES freetype PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH) #zlib, bzip2, libpng
@@ -169,6 +173,8 @@ find_library(EXPAT_RELEASE NAMES expat PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_D
 find_library(EXPAT_DEBUG NAMES expat PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
 
 #Dependent libraries
+find_library(ZSTD_RELEASE NAMES zstd zstd_static PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
+find_library(ZSTD_DEBUG NAMES zstdd zstd_staticd PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
 find_library(BZ2_RELEASE bz2 PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
 find_library(BZ2_DEBUG bz2 bz2d PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
 find_library(SSL_RELEASE ssl ssleay32 PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
@@ -197,6 +203,8 @@ set(RELEASE_OPTIONS
             "ICU_LIBS=${ICU_RELEASE}"
             "QMAKE_LIBS_PRIVATE+=${BZ2_RELEASE}"
             "QMAKE_LIBS_PRIVATE+=${LIBPNG_RELEASE}"
+            "QMAKE_LIBS_PRIVATE+=${ICU_RELEASE}"
+            "QMAKE_LIBS_PRIVATE+=${ZSTD_RELEASE}"
             )
 set(DEBUG_OPTIONS
             "LIBJPEG_LIBS=${JPEG_DEBUG}"
@@ -207,6 +215,8 @@ set(DEBUG_OPTIONS
             "ICU_LIBS=${ICU_DEBUG}"
             "QMAKE_LIBS_PRIVATE+=${BZ2_DEBUG}"
             "QMAKE_LIBS_PRIVATE+=${LIBPNG_DEBUG}"
+            "QMAKE_LIBS_PRIVATE+=${ICU_DEBUG}"
+            "QMAKE_LIBS_PRIVATE+=${ZSTD_DEBUG}"
             )
 
 

--- a/scripts/cmake/vcpkg_build_qmake.cmake
+++ b/scripts/cmake/vcpkg_build_qmake.cmake
@@ -69,15 +69,38 @@ function(vcpkg_build_qmake)
     unset(_buildname)
 
     foreach(_buildname ${BUILDTYPES})
-        set(_BUILD_PREFIX "${_path_suffix_${_buildname}}")
-        vcpkg_add_to_path(PREPEND "${CURRENT_INSTALLED_DIR}${_BUILD_PREFIX}/bin")
-        vcpkg_add_to_path(PREPEND "${CURRENT_INSTALLED_DIR}${_BUILD_PREFIX}/lib")
+        set(_installed_prefix_ "${CURRENT_INSTALLED_DIR}${_path_suffix_${_buildname}}")
+        set(_installed_libpath_ "${_installed_prefix_}/lib/${VCPKG_HOST_PATH_SEPARATOR}${_installed_prefix_}/lib/manual-link/")
+
+        vcpkg_add_to_path(PREPEND "${_installed_prefix_}/bin")
+        vcpkg_add_to_path(PREPEND "${_installed_prefix_}/lib")
+
+        # We set LD_LIBRARY_PATH ENV variable to allow executing Qt tools (rcc,...) even with dynamic linking
+        if(CMAKE_HOST_UNIX)
+            if(DEFINED ENV{LD_LIBRARY_PATH})
+                set(_ld_library_path_defined_ TRUE)
+                set(_ld_library_path_backup_ $ENV{LD_LIBRARY_PATH})
+                set(ENV{LD_LIBRARY_PATH} "${_installed_libpath_}${VCPKG_HOST_PATH_SEPARATOR}${_ld_library_path_backup_}")
+            else()
+                set(_ld_library_path_defined_ FALSE)
+                set(ENV{LD_LIBRARY_PATH} "${_installed_libpath_}")
+            endif()
+        endif()
+        
         list(APPEND _csc_${_buildname}_TARGETS ${_csc_TARGETS})
         if(NOT _csc_SKIP_MAKEFILES)
             run_jom(qmake_all makefiles ${_short_name_${_buildname}})
         endif()
         run_jom("${_csc_${_buildname}_TARGETS}" ${_csc_BUILD_LOGNAME} ${_short_name_${_buildname}})
-        unset(_BUILD_PREFIX)
+
+        # Restore backup
+        if(CMAKE_HOST_UNIX)
+            if(_ld_library_path_defined_)
+                set(ENV{LD_LIBRARY_PATH} "${_ld_library_path_backup_}")                
+            else()
+                unset(ENV{LD_LIBRARY_PATH})
+            endif()
+        endif()
     endforeach()
 
     # Restore the original value of ENV{PATH}

